src/flb_http_client.c

```diff
@@ -1107,6 +1107,43 @@ int flb_http_proxy_auth(struct flb_http_client *c,
     return flb_http_add_auth_header(c, user, passwd, FLB_HTTP_HEADER_PROXY_AUTH);
 }
 
+int flb_http_bearer_auth(struct flb_http_client *c, const char *token)
+{
+    int ret;
```

Comment:
Could you change the names of these variables to clearly reflect their intention?

Additionally, could you change what currently is `h` so it's a `flb_sds_t` string and then use `flb_sds_printf` with a `Bearer %s` as format string (and obviously verify that `token` is properly initialize to an empty `flb_sds_t` if not provided)

Then we can add a `flb_sds_destroy` at the end and get rid of the bulk of the code in this function which will make it much smaller and "idiomatic".